### PR TITLE
chore: refactored code

### PR DESCRIPTION
**Pull request by xnacly**
see commits

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -3,23 +3,38 @@ import consola from "consola";
...
-  } else if (isPnpm) {
-    consola.info("Detected package manager: pnpm");
-    return "pnpm";
+  let pkgMgr = getPkgMgr(cwd);
+  if (pkgMgr != "none") {
+    consola.info(`Detected package manager: ${pkgMgr}`);
+    return pkgMgr;
   } else {
-    consola.error("Can't detect any package manager");
+    consola.warn("Can't detect a package manager, auto fallback to: 'npm'");
```
Comment (Intevel) [inline src/index.ts]: `consola.error`

File: src/index.ts [hunk 2]
```diff
@@ -3,23 +3,38 @@ import consola from "consola";
...
+    return "npm";
```
Comment (Intevel) [inline src/index.ts]: Make an option to activate/deactivate fallback.

**Review by Intevel — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (xnacly) [reply src/index.ts]: I specifically used warn.  A warning is smarter to indicate something went wrong, but due to the fallback the user can still be sure the package was installed. Throwing an error would suggest the installing failed. The fallback to npm is useful due to many VCS users ignoring the `package-lock.json` file, even when they are using npm as a pkg-manager.

File: src/index.ts (hunk 2 above)
Comment (xnacly) [reply src/index.ts]: How should that be implemented? like the optional `cwd` flag?

File: src/index.ts (hunk 1 above)
Comment (Intevel) [reply src/index.ts]: Yes, but I wanted to make sure that the fallback is in the interest of the package user. So please add an option to activate or deactivate it.

File: src/index.ts (hunk 2 above)
Comment (Intevel) [reply src/index.ts]: Yes

File: src/index.ts (hunk 1 above)
Comment (xnacly) [reply src/index.ts]: will do

**Review by Intevel — APPROVED** [review]
Thank you for this PR. Good work 💚

**Pull request merged**